File: app/src/main/java/org/metabrainz/mobile/presentation/features/adapters/ResultItemComparator.java
package org.metabrainz.mobile.presentation.features.adapters;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

public class ResultItemComparator extends DiffUtil.ItemCallback<ResultItem> {
    @Override
    public boolean areItemsTheSame(@NonNull ResultItem oldItem, @NonNull ResultItem newItem) {
        return oldItem == newItem;
    }

    @Override
    public boolean areContentsTheSame(@NonNull ResultItem oldItem, @NonNull ResultItem newItem) {
        return oldItem.getMBID().equalsIgnoreCase(newItem.getMBID());
    }
}

File: app/src/main/java/org/metabrainz/android/ui/adapters/ResultItemComparator.kt
package org.metabrainz.android.ui.adapters

import androidx.recyclerview.widget.DiffUtil

class ResultItemComparator : DiffUtil.ItemCallback<ResultItem>() {
    override fun areItemsTheSame(oldItem: ResultItem, newItem: ResultItem): Boolean {
        return oldItem === newItem
    }

    override fun areContentsTheSame(oldItem: ResultItem, newItem: ResultItem): Boolean {
        return oldItem.mBID.equals(newItem.mBID, ignoreCase = true)
    }
}